src/io.c

```diff
@@ -2523,6 +2526,9 @@ _dispatch_operation_perform(dispatch_operation_t op)
 						processed = 0;
 					}
 				}
+				if (bSuccess && processed == 0) {
+					err = EAGAIN;
```

Comment:
Does this not need a `goto error`?  This will currently change the path to `return DISPATCH_OP_DELIVER_AND_COMPLETE` rather than `return DISPATCH_OP_COMPLETE_RESUME` or `return DISPATCH_OP_RESUME`.

Reply:
Oops, yes good catch!